workflows/prognostic_c48_run/runtime/steppers/prescriber.py

```diff
@@ -0,0 +1,274 @@
+                all wrapper timesteps for which data is required; if not supplied,
+                defaults to downloading entire time dimension of `dataset_key`
+        """
+        self._config = config
+        self._communicator = communicator
+        self._rename: Mapping[
+            Hashable, Hashable
+        ] = config.rename if config.rename else {}
+        self._timesteps = timesteps
+        self._prescribed_ds: xr.Dataset = self._scatter_prescribed_ds()
+
+    def _scatter_prescribed_ds(self):
+        time_coord: Optional[xr.DataArray]
+        if self._communicator.rank == 0:
+            prescribed_ds, time_coord = _get_prescribed_ds(
+                self._config.dataset_key,
+                list(self._config.variables),
+                self._timesteps,
+                self._config.consolidated,
+            )
+            prescribed_ds = _quantity_state_to_ds(
+                self._communicator.scatter_state(_ds_to_quantity_state(prescribed_ds))
+            )
+        else:
+            prescribed_ds = _quantity_state_to_ds(self._communicator.scatter_state())
+            time_coord = None
+        time_coord = self._communicator.comm.bcast(time_coord, root=0)
+        prescribed_ds = prescribed_ds.assign_coords({"time": time_coord})
+        return prescribed_ds
+
+    def __call__(self, time, state):
+        diagnostics: Diagnostics = {}
+        prescribed_timestep: xr.Dataset = _cast_to_double(
+            _add_net_shortwave(self._prescribed_ds.sel(time=time))
+        )
+        state_updates: State = self._rename_state(
+            {name: prescribed_timestep[name] for name in prescribed_timestep.data_vars}
+        )
+        for name in state_updates.keys():
+            diagnostics[name] = state_updates[name]
+        tendency: Tendencies = {}
+        return tendency, diagnostics, state_updates
+
+    def _rename_state(self, state: State) -> State:
+        new_state: State = {}
+        for name in state.keys():
+            if name in self._rename.keys():
+                new_state[self._rename[name]] = state[name]
+            else:
+                new_state[name] = state[name]
+        return new_state
+
+    def get_diagnostics(self, state, tendency):
+        return {}
+
+    def get_momentum_diagnostics(self, state, tendency):
+        return {}
+
+
+def _get_prescribed_ds(
+    dataset_key: str,
+    variables: Sequence[str],
+    timesteps: Optional[Sequence[cftime.DatetimeJulian]],
+    consolidated: bool = True,
+) -> Tuple[xr.Dataset, xr.DataArray]:
+    logger.info(f"Setting up dataset for state setting: {dataset_key}")
+    ds = _open_ds(dataset_key, consolidated)
+    ds = _remove_name_suffix(ds)
+    ds = get_variables(ds, variables)
+    ds = _rename_dims(ds)
+    ds = _set_missing_units_attr(ds)
+    ds = _round_time_coord(ds)
+    if timesteps is not None:
+        ds = ds.sel(time=timesteps)
+    time_coord = ds.coords["time"]
+    return ds.drop_vars(names="time").load(), time_coord
```

Comment:
This is a good strategic question...I think the approach here is probably only for setting 2D variables, and I'm not sure we have yet a use case for setting 3D variables. I'm hesitant to introduce another I/O call every timestep, since this already doesn't work all that great for the nudge-to-fine I/O, though I agree that with appropriate chunking maybe this is feasible.

Reply:
On the other hand if we're running relatively short segmented runs then it's less of a problem to download everything once.

Reply:
Leaving this as is as discussed